From 117a23131e8d10e16249ac8050b32db2cdaf1020 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Wed, 24 Jan 2018 10:49:10 +0800
Subject: [PATCH] Revert "asm-generic: fncpy: Add function copying macros"

This reverts commit 49d083bd72f8fe93b1d76078be1da897f5e41e98.

The patch is part of PIE, we do not need it anymore.

Change-Id: Iedc231105fa18c3cadd2cfd023c451c40570be96
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 arch/alpha/include/asm/fncpy.h      |   1 -
 arch/arc/include/asm/fncpy.h        |   1 -
 arch/arm64/include/asm/fncpy.h      |   1 -
 arch/avr32/include/asm/fncpy.h      |   1 -
 arch/blackfin/include/asm/fncpy.h   |   1 -
 arch/c6x/include/asm/fncpy.h        |   1 -
 arch/cris/include/asm/fncpy.h       |   1 -
 arch/frv/include/asm/fncpy.h        |   1 -
 arch/h8300/include/asm/fncpy.h      |   1 -
 arch/hexagon/include/asm/fncpy.h    |   1 -
 arch/ia64/include/asm/fncpy.h       |   1 -
 arch/m32r/include/asm/fncpy.h       |   1 -
 arch/m68k/include/asm/fncpy.h       |   1 -
 arch/metag/include/asm/fncpy.h      |   1 -
 arch/microblaze/include/asm/fncpy.h |   1 -
 arch/mips/include/asm/fncpy.h       |   1 -
 arch/mn10300/include/asm/fncpy.h    |   1 -
 arch/openrisc/include/asm/fncpy.h   |   1 -
 arch/parisc/include/asm/fncpy.h     |   1 -
 arch/powerpc/include/asm/fncpy.h    |   1 -
 arch/s390/include/asm/fncpy.h       |   1 -
 arch/score/include/asm/fncpy.h      |   1 -
 arch/sh/include/asm/fncpy.h         |   1 -
 arch/sparc/include/asm/fncpy.h      |   1 -
 arch/tile/include/asm/fncpy.h       |   1 -
 arch/um/include/asm/fncpy.h         |   1 -
 arch/unicore32/include/asm/fncpy.h  |   1 -
 arch/x86/include/asm/fncpy.h        |   1 -
 arch/xtensa/include/asm/fncpy.h     |   1 -
 include/asm-generic/fncpy.h         | 104 ----------------------------
 30 files changed, 133 deletions(-)
 delete mode 100644 arch/alpha/include/asm/fncpy.h
 delete mode 100644 arch/arc/include/asm/fncpy.h
 delete mode 100644 arch/arm64/include/asm/fncpy.h
 delete mode 100644 arch/avr32/include/asm/fncpy.h
 delete mode 100644 arch/blackfin/include/asm/fncpy.h
 delete mode 100644 arch/c6x/include/asm/fncpy.h
 delete mode 100644 arch/cris/include/asm/fncpy.h
 delete mode 100644 arch/frv/include/asm/fncpy.h
 delete mode 100644 arch/h8300/include/asm/fncpy.h
 delete mode 100644 arch/hexagon/include/asm/fncpy.h
 delete mode 100644 arch/ia64/include/asm/fncpy.h
 delete mode 100644 arch/m32r/include/asm/fncpy.h
 delete mode 100644 arch/m68k/include/asm/fncpy.h
 delete mode 100644 arch/metag/include/asm/fncpy.h
 delete mode 100644 arch/microblaze/include/asm/fncpy.h
 delete mode 100644 arch/mips/include/asm/fncpy.h
 delete mode 100644 arch/mn10300/include/asm/fncpy.h
 delete mode 100644 arch/openrisc/include/asm/fncpy.h
 delete mode 100644 arch/parisc/include/asm/fncpy.h
 delete mode 100644 arch/powerpc/include/asm/fncpy.h
 delete mode 100644 arch/s390/include/asm/fncpy.h
 delete mode 100644 arch/score/include/asm/fncpy.h
 delete mode 100644 arch/sh/include/asm/fncpy.h
 delete mode 100644 arch/sparc/include/asm/fncpy.h
 delete mode 100644 arch/tile/include/asm/fncpy.h
 delete mode 100644 arch/um/include/asm/fncpy.h
 delete mode 100644 arch/unicore32/include/asm/fncpy.h
 delete mode 100644 arch/x86/include/asm/fncpy.h
 delete mode 100644 arch/xtensa/include/asm/fncpy.h
 delete mode 100644 include/asm-generic/fncpy.h

diff --git a/arch/alpha/include/asm/fncpy.h b/arch/alpha/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/alpha/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/arc/include/asm/fncpy.h b/arch/arc/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/arc/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/arm64/include/asm/fncpy.h b/arch/arm64/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/arm64/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/avr32/include/asm/fncpy.h b/arch/avr32/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/avr32/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/blackfin/include/asm/fncpy.h b/arch/blackfin/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/blackfin/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/c6x/include/asm/fncpy.h b/arch/c6x/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/c6x/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/cris/include/asm/fncpy.h b/arch/cris/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/cris/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/frv/include/asm/fncpy.h b/arch/frv/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/frv/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/h8300/include/asm/fncpy.h b/arch/h8300/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/h8300/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/hexagon/include/asm/fncpy.h b/arch/hexagon/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/hexagon/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/ia64/include/asm/fncpy.h b/arch/ia64/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/ia64/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/m32r/include/asm/fncpy.h b/arch/m32r/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/m32r/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/m68k/include/asm/fncpy.h b/arch/m68k/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/m68k/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/metag/include/asm/fncpy.h b/arch/metag/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/metag/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/microblaze/include/asm/fncpy.h b/arch/microblaze/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/microblaze/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/mips/include/asm/fncpy.h b/arch/mips/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/mips/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/mn10300/include/asm/fncpy.h b/arch/mn10300/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/mn10300/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/openrisc/include/asm/fncpy.h b/arch/openrisc/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/openrisc/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/parisc/include/asm/fncpy.h b/arch/parisc/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/parisc/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/powerpc/include/asm/fncpy.h b/arch/powerpc/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/powerpc/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/s390/include/asm/fncpy.h b/arch/s390/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/s390/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/score/include/asm/fncpy.h b/arch/score/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/score/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/sh/include/asm/fncpy.h b/arch/sh/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/sh/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/sparc/include/asm/fncpy.h b/arch/sparc/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/sparc/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/tile/include/asm/fncpy.h b/arch/tile/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/tile/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/um/include/asm/fncpy.h b/arch/um/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/um/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/unicore32/include/asm/fncpy.h b/arch/unicore32/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/unicore32/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/x86/include/asm/fncpy.h b/arch/x86/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/x86/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/arch/xtensa/include/asm/fncpy.h b/arch/xtensa/include/asm/fncpy.h
deleted file mode 100644
index ee4741c11b1d..000000000000
--- a/arch/xtensa/include/asm/fncpy.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <asm-generic/fncpy.h>
diff --git a/include/asm-generic/fncpy.h b/include/asm-generic/fncpy.h
deleted file mode 100644
index 1a252828a456..000000000000
--- a/include/asm-generic/fncpy.h
+++ /dev/null
@@ -1,104 +0,0 @@
-/*
- * include/asm-generic/fncpy.h - helper macros for function body copying
- *
- * Copyright (C) 2011 Linaro Limited
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
- */
-
-/*
- * These macros are intended for use when there is a need to copy a low-level
- * function body into special memory.
- *
- * For example, when reconfiguring the SDRAM controller, the code doing the
- * reconfiguration may need to run from SRAM.
- *
- * NOTE: that the copied function body must be entirely self-contained and
- * position-independent in order for this to work properly.
- *
- * Typical usage example:
- *
- * extern int f(args);
- * extern uint32_t size_of_f;
- * int (*copied_f)(args);
- * void *sram_buffer;
- *
- * copied_f = fncpy(sram_buffer, &f, size_of_f);
- *
- * ... later, call the function: ...
- *
- * copied_f(args);
- *
- * The size of the function to be copied can't be determined from C:
- * this must be determined by other means, such as adding assmbler directives
- * in the file where f is defined.
- */
-
-#ifndef __ASM_GENERIC_FNCPY_H
-#define __ASM_GENERIC_FNCPY_H
-
-#include <linux/types.h>
-#include <linux/string.h>
-
-#include <asm/bug.h>
-#include <asm/cacheflush.h>
-
-/*
- * Minimum alignment requirement for the source and destination addresses
- * for function copying.
- */
-#ifndef ARCH_FNCPY_ALIGN
-#define ARCH_FNCPY_ALIGN	0
-#endif
-
-#define ARCH_FNCPY_MASK		((1 << (ARCH_FNCPY_ALIGN)) - 1)
-
-#ifndef fnptr_to_addr
-#define fnptr_to_addr(funcp) ({						\
-	(uintptr_t) (funcp);						\
-})
-#endif
-
-#ifndef fnptr_translate
-#define fnptr_translate(orig_funcp, new_addr) ({			\
-	(typeof(orig_funcp)) (new_addr);				\
-})
-#endif
-
-/* Ensure alignment of source and destination addresses */
-#ifndef fn_dest_invalid
-#define fn_dest_invalid(funcp, dest_buf) ({				\
-	uintptr_t __funcp_address;					\
-									\
-	__funcp_address = fnptr_to_addr(funcp);				\
-									\
-	((uintptr_t)(dest_buf) & ARCH_FNCPY_MASK) ||			\
-		(__funcp_address & ARCH_FNCPY_MASK);			\
-})
-#endif
-
-#ifndef fncpy
-#define fncpy(dest_buf, funcp, size) ({					\
-	BUG_ON(fn_dest_invalid(funcp, dest_buf));			\
-									\
-	memcpy(dest_buf, (void const *)(funcp), size);			\
-	flush_icache_range((unsigned long)(dest_buf),			\
-		(unsigned long)(dest_buf) + (size));			\
-									\
-	fnptr_translate(funcp, dest_buf);				\
-})
-#endif
-
-#endif /* !__ASM_GENERIC_FNCPY_H */
-
-- 
2.35.3

